crates/proto/src/rr/domain/name.rs

```diff
@@ -161,9 +169,17 @@ impl Name {
             return Err(format!("error converting some labels: {:?}", errors).into());
         };
 
+        let mut label_ends = TinyVec::new();
```

Comment:
Is there some way we can deduplicate the append label logic between the method and the constructor (without affecting performance or otherwise impacting legibility of the code)?

Reply:
Whoops, I was meaning to change this. I don't think there's any reason to expect a perf impact (not that I have a benchmark, so who knows) because the only change is adding a length check that should always pass. Not free, but small compared to creating an `Arc` for every `Label`.

It would probably be nice to use `Label` more in this file for cleanliness and to hoist some length checks, but I'd prefer to keep the change set smaller and tackle other optimization opportunities first.

Reply:
Right, keeping the change set small at first is my preferred strategy, too.